### PR TITLE
📖 Rewrite the code docs

### PR DESCRIPTION
**Pull request by ElRaptorus**
**Changes:**

Change the code documentations to make it more comprehensive.
Most docs were taken from their Consumer API counterpart, to ensure synchronicity.

1. Add `@throws` sections to all API functions
2. `@returns` now only describes the values that the function returns, not the Promise that wraps the function. That is already covered by the `@async` tag.
3. Remove some docs that serve no useful purpose, such as `The Id.` or `The list.`.
4. Unify usages of terms such as `ID` or `FormFields`.

PR: #45

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.).
- [x] I've rebased the `develop` branch with my branch before finishing this PR.
- [x] I've **summarized all changes** in a list above.
- [x] I've mentioned all **PRs, which relate to this one**.
- [x] I've prefixed my Pull Request title is according to [gitmoji guide](https://gitmoji.carloscuesta.me/).

### CODE REVIEW

File: src/data_models/user_task/user_task_form_field_type.ts [hunk 1]
```diff
@@ -2,10 +2,28 @@
...
+   * The FormField is long-type numerical.
+   */
   long = 'long',
+  /**
+   * The FormField is a number.
+   */
   number = 'number',
+  /**
+   * The FormField is a string.
+   */
```
Comment (cmg-dev) [inline src/data_models/user_task/user_task_form_field_type.ts]: Well following the same argument about useful comments, this is probably too much, too.

```
export enum UserTaskFormFieldType {
	  /**
	   * The FormField is a boolean.
	   */
	  boolean = 'boolean',
```

Basically translates to:

Form-field types knows a boolean and it is a boolean 😲 

maybe removing it, is more simple.

File: src/data_models/user_task/user_task_enum_value.ts [hunk 2]
```diff
@@ -1,7 +1,14 @@
...
+ * about the values contained within that enumeration.
  */
 export class UserTaskEnumValue {
+  /**
+   * The unique identifier of the value.
+   */
   public id: string;
+  /**
+   * The name of the value.
+   */
```
Comment (cmg-dev) [inline src/data_models/user_task/user_task_enum_value.ts]: same here.

File: src/data_models/user_task/user_task_config.ts [hunk 3]
```diff
@@ -4,6 +4,20 @@ import {UserTaskFormField} from './user_task_form_field';
  * Contains information about a UserTasks configuration.
  */
 export class UserTaskConfig {
+  /**
+   * A list of accessible FormFields for the UserTask.
```
Comment (cmg-dev) [inline src/data_models/user_task/user_task_config.ts]: this is a useful comment.

File: src/data_models/user_task/user_task_config.ts [hunk 4]
```diff
@@ -4,6 +4,20 @@ import {UserTaskFormField} from './user_task_form_field';
...
+   */
   public formFields: Array<UserTaskFormField> = [];
+  /**
+   * The perferred type of control for the FormField.
+   */
   public preferredControl?: string;
+  /**
+   * Optional: A description of this UserTask.
```
Comment (cmg-dev) [inline src/data_models/user_task/user_task_config.ts]: ... this maybe not.

File: src/data_models/user_task/user_task_config.ts [hunk 5]
```diff
@@ -4,6 +4,20 @@ import {UserTaskFormField} from './user_task_form_field';
...
+   */
+  public description?: string;
+  /**
+   * Optional: A message to show when the UserTask has been finished.
```
Comment (cmg-dev) [inline src/data_models/user_task/user_task_config.ts]: this too. will it help me, if I want to work with the code base?

File: src/data_models/user_task/user_task.ts [hunk 6]
```diff
@@ -1,20 +1,36 @@
-import {IIdentity} from '@essential-projects/iam_contracts';
-
 import {UserTaskConfig} from './user_task_config';
 
 /**
- * Describes a user task that belongs to an active ProcessInstance.
+ * Describes a waiting UserTask.
```
Comment (cmg-dev) [inline src/data_models/user_task/user_task.ts]: Only waiting UserTasks?

What is a waiting UserTask anyway? <- this is something a user might ask.

File: src/data_models/user_task/user_task.ts [hunk 7]
```diff
@@ -1,20 +1,36 @@
...
  */
 export class UserTask {
+  /**
+   * The model ID of the UserTask, as it is declared in the ProcessModel.
+   */
   public id: string;
-  public flowNodeInstanceId?: string;
+  /**
+   * The name of the UserTask, as it is declared in the ProcessModel.
```
Comment (cmg-dev) [inline src/data_models/user_task/user_task.ts]: to reference the ProcessModel is a good idea.

then it clearly says, that this is connected to the .bpmn file. <3

File: src/data_models/user_task/user_task.ts [hunk 8]
```diff
@@ -1,20 +1,36 @@
...
+   */
   public name: string;
+  /**
+   * The instance ID of the UserTask.
```
Comment (cmg-dev) [inline src/data_models/user_task/user_task.ts]: this is another example of probably not useful comments.

File: src/data_models/process_models/process_start_request_payload.ts [hunk 9]
```diff
@@ -1,21 +1,21 @@
 /**
- * Describes the payload that a process model can be started with.
+ * Describes the payload that a ProcessModel can be started with.
  */
 export class ProcessStartRequestPayload {
   /**
-   * Optional: If set, the process engine will use this as the id for the
-   * correlation in which the process model will be executed.
-   * If not provided, the process engine will generated a correlation id by
+   * Optional: If set, the process engine will use this as the ID for the
```
Comment (cmg-dev) [inline src/data_models/process_models/process_start_request_payload.ts]:
```suggestion
   * Optional: If set, the Process-Engine will use this as the ID for the
```

Process-Engine is still our product and it is its name.

Comment (cmg-dev) [reply src/data_models/process_models/process_start_request_payload.ts]: Or if you want to reference a generic engine, use 

> workflow engine

instead.

File: src/data_models/manual_task/manual_task.ts [hunk 10]
```diff
@@ -1,14 +1,33 @@
...
+   * The model ID of the ManualTask, as it is declared in the ProcessModel.
+   */
   public id: string;
-  public flowNodeInstanceId?: string;
+  /**
+   * The name of the ManualTask, as it is declared in the ProcessModel.
+   */
   public name: string;
+  /**
+   * The instance ID of the ManualTask.
```
Comment (cmg-dev) [inline src/data_models/manual_task/manual_task.ts]: same here.

File: src/data_models/event/event.ts [hunk 11]
```diff
@@ -1,25 +1,40 @@
...
+   * The instance ID of the event.
    */
-  public eventName: string;
+  public flowNodeInstanceId?: string;
+  /**
+   * The ID of the Correlation that the event belongs to.
+   */
+  public correlationId: string;
+  /**
+   * The ID of the ProcessModel that the event belongs to.
```
Comment (cmg-dev) [inline src/data_models/event/event.ts]: same.

File: src/data_models/user_task/user_task_form_field_type.ts (hunk 1 above)
Comment (ElRaptorus) [reply src/data_models/user_task/user_task_form_field_type.ts]: True. 🙂

File: src/data_models/process_models/process_start_request_payload.ts (hunk 9 above)
Comment (ElRaptorus) [reply src/data_models/process_models/process_start_request_payload.ts]: missed that one.

File: src/data_models/user_task/user_task.ts (hunk 8 above)
Comment (ElRaptorus) [reply src/data_models/user_task/user_task.ts]: Not sure about this one.
I kept it intentionally, to keep the difference between ModelId and FlowNodeInstanceId visible. 
I think if the comment about the ModelId makes sense, then this one does as well.

File: src/data_models/manual_task/manual_task.ts (hunk 10 above)
Comment (ElRaptorus) [reply src/data_models/manual_task/manual_task.ts]: See above.

**Pull request merged**